File: desktop/src/types/call.ts
export type CallStatus = "ringing" | "in_progress" | "ended" | "missed" | "declined";
export type CallType = "audio" | "video";

export interface Call {
  id: string;
  channel_id: string;
  initiator_id: string;
  call_type: CallType;
  status: CallStatus;
  accepted_by?: string;
  started_at?: string;
  ended_at?: string;
  created_at: string;
}

export interface CallParticipant {
  call_id: string;
  user_id: string;
  joined_at?: string;
  left_at?: string;
}

export interface ICEServerConfig {
  urls: string[];
  username?: string;
  credential?: string;
}

export interface RTCConfig {
  ice_servers: ICEServerConfig[];
  enabled: boolean;
}

File: server/internal/model/call.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type CallStatus string
type CallType string

const (
	CallStatusRinging    CallStatus = "ringing"
	CallStatusInProgress CallStatus = "in_progress"
	CallStatusEnded      CallStatus = "ended"
	CallStatusMissed     CallStatus = "missed"
	CallStatusDeclined   CallStatus = "declined"
)

const (
	CallTypeAudio CallType = "audio"
	CallTypeVideo CallType = "video"
)

type Call struct {
	ID          uuid.UUID  `json:"id"`
	ChannelID   uuid.UUID  `json:"channel_id"`
	InitiatorID uuid.UUID  `json:"initiator_id"`
	CallType    CallType   `json:"call_type"`
	Status      CallStatus `json:"status"`
	StartedAt   *time.Time `json:"started_at,omitempty"`
	EndedAt     *time.Time `json:"ended_at,omitempty"`
	CreatedAt   time.Time  `json:"created_at"`
}

type CallParticipant struct {
	CallID   uuid.UUID  `json:"call_id"`
	UserID   uuid.UUID  `json:"user_id"`
	JoinedAt *time.Time `json:"joined_at,omitempty"`
	LeftAt   *time.Time `json:"left_at,omitempty"`
}

type InitiateCallRequest struct {
	ChannelID uuid.UUID `json:"channel_id" validate:"required"`
	CallType  CallType  `json:"call_type" validate:"required,oneof=audio video"`
}
